python/triton/language/semantic.py

```diff
@@ -1394,7 +1398,8 @@ def dot(lhs: tl.tensor, rhs: tl.tensor, acc: tl.tensor, input_precision: Optiona
 
 
 def where(condition: tl.tensor, x: tl.tensor, y: tl.tensor, builder: ir.builder) -> tl.tensor:
-    condition = cast(condition, tl.int1, builder)
+    if condition.dtype != tl.int1:
+        raise ValueError(f"tl.where: condition must be a boolean. Got {condition.dtype}")
```

Comment:
BTW should we start with a warning and plan to make it an error in future? I know that triton hasn't cared to much about breaking BC in the past but it isn't very friendly to users.

Reply:
cc @ThomasRaoux @ptillet

Reply:
that's a good point. Starting with a warning makes sense to me.